.github/workflows/sonar.yml

```diff
@@ -0,0 +1,34 @@
+name: Sonar
+
+on:
+  pull_request:
```

Comment:
You won't be able to access secrets from pull requests, so it doesn't make sense to have sonar run on pull requests like this.

You could leave this but it will only invoke sonar for pull requests from contributors against master.

Reply:
For reference: https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#using-encrypted-secrets-in-a-workflow

> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

Reply:
I don't think a pull request is a fork repository, it's just a branch.. But I might be wrong. If the secret change fix the issue I will test with my next pull request.

Reply:
Oh I think I get what you are saying. I change it to "push", everything that is push on this repository will be scanned. Might change this later.